homeassistant/components/overkiz/climate_entities/atlantic_electrical_towel_dryer.py

```diff
@@ -0,0 +1,118 @@
+"""Support for Atlantic Electrical Towel Dryer."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    PRESET_BOOST,
+    PRESET_NONE,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.components.overkiz.coordinator import OverkizDataUpdateCoordinator
+from homeassistant.components.overkiz.entity import OverkizEntity
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+PRESET_DRYING = "drying"
+
+OVERKIZ_TO_HVAC_MODE: dict[str, str] = {
+    OverkizCommandParam.EXTERNAL: HVACMode.HEAT,  # manu
+    OverkizCommandParam.INTERNAL: HVACMode.AUTO,  # prog
+    OverkizCommandParam.STANDBY: HVACMode.OFF,
+}
+HVAC_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODE.items()}
+
+OVERKIZ_TO_PRESET_MODE: dict[str, str] = {
+    OverkizCommandParam.PERMANENT_HEATING: PRESET_NONE,
+    OverkizCommandParam.BOOST: PRESET_BOOST,
+    OverkizCommandParam.DRYING: PRESET_DRYING,
+}
+
+PRESET_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODE.items()}
+
+
+class AtlanticElectricalTowelDryer(OverkizEntity, ClimateEntity):
+    """Representation of Atlantic Electrical Towel Dryer."""
+
+    _attr_hvac_modes = [*HVAC_MODE_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODE_TO_OVERKIZ]
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(7)
+
+        self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+
+        # Not all AtlanticElectricalTowelDryer models support presets, thus we need to check if the command is available
+        if self.executor.has_command(OverkizCommand.SET_TOWEL_DRYER_TEMPORARY_STATE):
+            self._attr_supported_features += ClimateEntityFeature.PRESET_MODE
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if OverkizState.CORE_OPERATING_MODE in self.device.states:
+            return OVERKIZ_TO_HVAC_MODE[
+                cast(str, self.executor.select_state(OverkizState.CORE_OPERATING_MODE))
+            ]
```

Comment:
```suggestion
            state = self.executor.select_state(OverkizState.CORE_OPERATING_MODE)
            return OVERKIZ_TO_HVAC_MODE[cast(str, state)]
```

Reply:
I tried to adhere to the style we use in the other Overkiz entities as well; I don't think that first saving it in a local variable will make it more clear, imo it only adds another layer of complexity. 

I would rather implement the `select_as_string` eventually and make it more clear, as we discussed a while ago. I think we had a draft for that, perhaps we should try to move that one forward?

Reply:
OK to start that discussing again and keeping this for now?